src/DynamoCore/Configuration/IPreferences.cs

```diff
@@ -166,6 +166,22 @@ public interface IRenderPrecisionPreference
         int RenderPrecision { get; set; }
     }
 
+    /// <summary>
+    /// Temporary interface to avoid breaking changes.
+    /// TODO: Merge with IPreferences for 3.0 (DYN-1699)
+    /// </summary>
+    internal interface IDisablePackageLoadingPreferences
+    {
+        /// <summary>
+        /// If enabled Dynamo Standard Library packages will not be loaded.
+        /// </summary>
+        bool DisableStandardLibrary { get; set; }
+        /// <summary>
+        /// If enabled user's custom package locations will not be loaded.
+        /// </summary>
+        bool DisableCustomPackageLocations { get; set; }
+    }
```

Comment:
Could these 2 bools be turned into a single list like `IEnumerable<string> DisablePackageLocations ` ?
API user could add std lib location and customPackagesLocation to this list.

Reply:
This doesn't work well from a security/hardening standpoint because you would need to know all custom package paths to disable them - unless you mean literally adding the strings
"standardLibrary" and "customPackagePaths" ?

Reply:
I was actually thinking about having a more generic API that could handle future changes to the package locations.
If we add more locations or remove one of them...we will have to deal with these booleans for a while...

Reply:
To that end I think the the bit flags or an IEnumerable<string> where each string is a package location type(not path) - could work.
doyou prefer one or the other?

Reply:
I was originally thinking of list of paths...but you bring up a good point...
I think the current way (booleans) is easier to use/understand than enums or list of types.